client-app/src/desktop/AppModel.ts

```diff
@@ -44,18 +44,8 @@ export class AppModel extends HoistAppModel {
         ]
     });
 
-    static override async preAuthAsync() {
-        const whitelist = ['toolbox.xh.io', 'toolbox-dev.xh.io', 'localhost'];
-        if (whitelist.includes(window.location.hostname)) {
-            await XH.installServicesAsync(OauthService);
-        } else {
-            // We are not running in an environment that is supported by auth0
-            // (presumably `yarn startOnDevice`), so fall back to username / password prompt.
-            XH.appSpec.isSSO = false;
-        }
-    }
```

Comment:
I apologize, but this bit where we detect the common case where we don't expect Auth0 to work (non-whitelisted hostname) and automatically fallback to the form-based login is quite nice - I had forgotten we had this in here and that it provided a pretty good solution to the "startOnDevice" problem (although not the "auth0 is misconfigured" problem).

Could we mix this in to the new preAuth check, so we get the best of all worlds?  That way a user who is running on a non-standard hostname and has the bootstrapAdminUser set could get the working form-based auth directly from the app they meant to load.  The bootstrapLogin would then be a true backup in the event that auth0 is broken.